### PR TITLE
Feature/fix build

### PR DESCRIPTION
**Pull request by roykho**
So this came out of a hack week project that I've started. It is basically a progression to make our build process and local development more developer friendly.

**Features:**
* All min files are now properly ignored so that you won't get any untracked files warning when working locally. This is evident when using `grunt watch` and other scripts.
* All JS files are now included in the Uglify process of Grunt so that it is consistent with other JS files we currently Uglify. This also prevents an accidental change in the main JS file that may have forgotten to minify.
* Add woocommerce.zip to gitignore as this comes up as untracked file when running the build script locally.
* Added `npm run test:unit` to run our unit tests.

**Removed:**
* Removed all the JS min files as these will now all be built with the build/watch script.

**Testing**
* Remove all min files and run `npm run build-watch` and ensure all min files are created.
* Remove all min files and run `npm run build` and ensure all min files are included and in the zip.
* Ensure we don't have any missing min files for each vendor packages we're using.
* Ensure you're not seeing any untracked files for the JS and zip files.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -26,6 +26,25 @@ none
 # Minified JS
 /assets/js/admin/*.min.js
 /assets/js/frontend/*.min.js
+/assets/js/accounting/*.min.js
```
Comment (vedanshujain) [inline .gitignore:29]: Thanks, this looks much needed, what do you think about ignoring all min files by using something like `assets/**/*-min.js`

File: package.json [hunk 2]
```diff
@@ -28,6 +28,7 @@
     "test:e2e": "npm explore @woocommerce/e2e-environment -- npm run test:e2e",
     "test:e2e-debug": "npm explore @woocommerce/e2e-environment -- npm run test:e2e-debug",
     "test:e2e-dev": "npm explore @woocommerce/e2e-environment -- npm run test:e2e-dev",
+    "test:unit": "./vendor/bin/phpunit",
```
Comment (vedanshujain) [inline package.json:31]: also add `-c ./phpunit.xml` to make it more accurate

**Review by vedanshujain — COMMENTED** [review]
Looks good to me! Left a couple of minor comments. I tested that all files are build by:

1. Running `npm run build:assets` on a temp branch (forked from master), going to to .gitignore to include back all js files and commit it.
2. Removed all js min files using `rm assets/**/*.min.js`. Verified files are removed using git status
3. Checked out this branch and ran `npm run build:assets` again
4. Checkout temp branch again and verify that working tree is clean.

will approve after addressing of those comments

File: .gitignore (hunk 1 above)
Comment (roykho) [reply .gitignore:29]: Yes I thought about that at first however I then see this https://github.com/woocommerce/woocommerce/blob/master/.gitignore#L27-L28

So I was just following the existing pattern. I thought maybe we wanted to be verbose here. But I am ok to change it.

**Review by vedanshujain — APPROVED** [review]
Looks good! Test also passing on re-run.

**Pull request merged**